LayoutTests/imported/w3c/web-platform-tests/service-workers/service-worker/fetch-event.https.html

```diff
@@ -147,6 +147,28 @@
             'Service Worker should respond with an empty resulting client id for subresource requests');
         });
   }, 'Service Worker responds to fetch event with the correct resulting client id');
+  
+  promise_test(t => {
+      const page_url = 'resources/simple.html?replacesClientId';
```

Comment:
This is a same origin navigation.
I wonder what we should do for cross origin navigations. It might be good to write a test, maybe for cross origin redirection as well.
Are we expected to expose cross origin client IDs given the service worker cannot do a lot of things with it?

Reply:
>  It might be good to write a test, maybe for cross origin redirection as well.

Ok I can add one. 

> Are we expected to expose cross origin client IDs given the service worker cannot do a lot of things with it?

I think so yes, since I cannot find anything to the contrary in the spec. cc @annevk 

The worker could use the id to clear some data related to the client that is about to go away?

Reply:
Hmm. A `Client` object is static right and represents what the service worker intercepts? How could it represent a cross-origin environment?

I guess that's possible with `openWindow()`? That seems wrong-ish.

Reply:
FYI had a look at https://bugs.webkit.org/attachment.cgi?id=449475&action=diff, but it seems the previously implemented `targetClientId` was only implemented as a default empty string, it wasn't actually set yet. So no guidance there. 

I think `openWindow()` should show an empty `replaclesClientId` because it loads the url in a newly created top-level bc, so the one being replaced would be "about:blank". 

It seems the cross-origin case would rather be about a client navigating an existing iframe across origin? If the resulting client id is exposed in that case, then the one being replaced could be exposed as well?

Reply:
When A embeds A1 and A1 is navigated to B, A's service worker would only service A1, not B. B would have its own service worker. I don't think we'd expose a client of B to A? (If we do, I think that would be a bug, probably a security bug.)

Reply:
In case of A1 -> B cross-origin navigation, we would start exposing A1 id to B service worker.
I am not sure there is a security issue here but I don't really understand what B service worker would do with A1 id in case of cross-origin navigation.
Are we already exposing cross origin env IDs elsewhere? If this is new, why should we start doing it?

Reply:
We don't really expose any information cross-origin currently, not even URL. E.g., `location.href` only has a cross-origin setter. So that definitely seems like a bug.

Reply:
> A1 -> B cross-origin navigation, we would start exposing A1 id to B service worker

How does `resultingClientId` currently deals with the mirror of that example(the future id of B being exposed in the fetch event to A's sw, with A embedding A1)?

Reply:
> A1 -> B cross-origin navigation, we would start exposing A1 id to B service worker

Also, if A1 is embedded as a frame in A, and they are same-origin, with A being a client to a sw. If A1 is navigated to a cross-origin destination B(through JS manipulated it in the context of A), then a fetch event will be delivered to the sw of A, and not to what will be the sw of B after the navigation completes, right? If that is correct, then the `replacesClientId` will not be cross-origin, it will be that of A1 prior to the navigation, and it will be delivered to the same origin sw of A. 

If A1 is cross origin to begin with, then it cannot be navigated by code in A(right?).

Reply:
Navigation loads are different from sub resource loads as they trigger matching service worker.
See https://w3c.github.io/ServiceWorker/#on-fetch-request-algorithm step 14.6 (vs step 15 for sub resources).
Navigation loads are always handled by same origin service workers, hence why resultingClientId is also always same origin.
In case of cross origin navigation redirections, a new resultingClientId will be generated. 

> By the way, is it to correct to say that the only use-case of replacesClientId is the navigation of an embedded frame, since in other cases clientId will always equal replacesClientId?

Top level frame navigations may expose replacesClientId.

Reply:
> Navigation loads are different from sub resource loads as they trigger matching service worker.

Thanks for the info! 

In that case, it appears to me that the `replacesClientId` will always be that of a client whose origin matches the worker receiving the fetch event.

Reply:
>  it appears to me that the replacesClientId will always be that of a client whose origin matches the worker receiving the fetch event. Except maybe in the cross-origin redirect case.

Can you clarify? My understanding is that `replacesClientId` would be cross origin in case of any cross origin navigation, except if the spec is filtering somehow exposure of the env id doing the navigation.

Reply:
Ok, sorry I was confused about the matching algorithm, though it would match on the worker of the origin where the navigation request originates, while it is in fact matching on the "target" origin. 

There is even an assertion in step 8.2 [the matching algorithm](https://w3c.github.io/ServiceWorker/#match-service-worker-registration) that the origin of the url of the navigation and the resulting matching scope are equal(added [here](https://github.com/w3c/ServiceWorker/issues/1118)). 

So, it feels as if [Step 24.3.9 of Handle Fetch](https://w3c.github.io/ServiceWorker/#handle-fetch) should include a same origin check and initialize `replacesClientId` to the empty string if it fails?

Could one use the request's client origin, and compare it with the request's url?

Reply:
Yes, that should do the trick spec wise, it would be good to discuss this with the service worker WG.
The thing to cover well in terms of WPT tests will be redirections, ideally both iframes and top level navigations.